src/compiler/checker.ts

```diff
@@ -38211,7 +38211,8 @@ namespace ts {
         function isReferencedAliasDeclaration(node: Node, checkChildren?: boolean): boolean {
             if (isAliasSymbolDeclaration(node)) {
                 const symbol = getSymbolOfNode(node);
-                if (symbol && getSymbolLinks(symbol).referenced) {
+                const links = symbol && getSymbolLinks(symbol);
+                if (links?.referenced || (compilerOptions.isolatedModules && links?.constEnumReferenced)) {

```

Comment:
This is incorrect fix. without `preserveConstEnums` alias is not treated as imported . the fix should be to error instead.

Reply:
Wait isolatedModules is suppose to treat it as `preserveConstEnums` ? 
We have descripency in how we handle thsese two flags and that should be fixed instead

Reply:
Yeah, see https://github.com/microsoft/TypeScript/blob/9dfbf07d8a63a8d6145cb67aa33d7486ad176152/src/compiler/transformers/ts.ts#L2317-L2321. I didn’t realize that either until #40499 pointed it out, but it seems to be intentional. That’s why I think this fix is ok, and making it an error needs a discussion. It seems weird, but there could be people relying on this behavior.

Reply:
i think this fix is not ok.. We should may be have function preserverConstEnums(options) = options.preserveConstEnums || options.isolatedModules;  and use it everywhere we intend to use options.preserverConstEnum to ensure that all the flag setting take place in either mode.

Reply:
Eg,. markAliasReferenced needs to handle that. that would be correct fix

Reply:
Yes, that would be a good addition.

Reply:
We also need a config error for `--isolatedModules` combined with `--preserveConstEnums=false`

Reply:
I've pushed a commit which adds the config error.

I don't know anything about the config system, or even really the compiler (I'm only here because this bug caused an outage), but.. would it not be easier to make `isolatedModules` just imply `--preserveConstEnums=true` at the config level, and leave everything else alone? There's a lot of references to both options, and some of them aren't obviously correct either way around.

Reply:
We tend not to make `compilerOptions` belie the actual config in case there’s a need to separate out what options the user actually specified later. I already searched through every usage of `preserveConstEnums` and verified they can pretty much all be replaced with `getPreserveConstEnums(compilerOptions)`. Note we do the same thing with `getAllowSyntheticDefaultImports(compilerOptions)`, `getEmitDeclarations(compilerOptions)`, and several others.

Reply:
Okay, understood.

Do you want me to have a go too, or do you have a commit to push?

Reply:
I don’t have a commit—if you want to do it, awesome; if this is getting to be more than you signed up for, I can finish it out.

Reply:
I've pushed that change, which doesn't seem to break any tests, and isn't sufficient to fix my new tests; we still need the other commits.